contracts/astroport-liquidity-helper/tests/astroport_integration_tests.rs

```diff
@@ -214,74 +217,110 @@ pub fn test_calc_xyk_balancing_swap() {
+    true
 )]
-#[test_case(
-    [Uint128::from(0u128), Uint128::from(1_000_000_000_000u128)],
-    [Uint128::from(1_000_000_000_000u128), Uint128::from(2_000_000_000_000u128)],
-    true;
-    "Test 4: 1:2 pool ratio, 0:1 ratio of assets"
+// Test 4: 1:2 pool ratio, 0:1 ratio of assets
+#[test_matrix(
+    [[Uint128::from(0u128), Uint128::from(1_000_000_000_000u128)]],
+    [[Uint128::from(1_000_000_000_000u128), Uint128::from(2_000_000_000_000u128)]],
+    [PairType::Xyk {},PairType::Stable {}, PairType::Custom("concentrated".to_string())],
+    true
+)]
+// Test 5: 1:1 pool ratio, 1:1 ratio of assets
+#[test_matrix(
+    [[Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)]],
+    [[Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)]],
+    [PairType::Xyk {},PairType::Stable {}, PairType::Custom("concentrated".to_string())],
+    true
+)]
+// Test 6: 1:1 pool ratio, 1:0 ratio of assets
+#[test_matrix(
+    [[Uint128::from(1_000_000_000u128), Uint128::from(0u128)]],
+    [[Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)]],
+    [PairType::Xyk {},PairType::Stable {}, PairType::Custom("concentrated".to_string())],
+    true
 )]
 #[test_case(
-    [Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)],
-    [Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)],
-    true;
-    "Test 5: 1:1 pool ratio, 1:1 ratio of assets"
+    [Uint128::from(0u128), Uint128::from(3564u128)],
+    [Uint128::from(3450765745u128), Uint128::from(12282531965699u128)],
+    PairType::Xyk {},
+    false;
+    "Test 7: Xyk amount of asset less than one microunit of other asset"
+)]
+// Test 7: Amount of asset would be less than one microunit of other asset if it were xyk
+#[test_matrix(
+    [[Uint128::from(0u128), Uint128::from(3564u128)]],
+    [[Uint128::from(3450765745u128), Uint128::from(12282531965699u128)]],
+    [PairType::Stable {  }, PairType::Custom("concentrated".to_string())],
+    true
 )]
 #[test_case(
     [Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)],
     [Uint128::from(0u128), Uint128::from(0u128)],
+    PairType::Xyk {},
     true
     => panics "No liquidity in pool";
-    "Test 6: 0:0 pool ratio, should fail with correct error"
+    "Test 8 Xyk: 0:0 pool ratio, should fail with correct error"
 )]
-#[test_case(
-    [Uint128::from(1_000_000_000_000u128), Uint128::from(0u128)],
-    [Uint128::from(1_000_000_000_000u128), Uint128::from(1_000_000_000_000u128)],
-    true;
-    "Test 7: 1:1 pool ratio, 1:0 ratio of assets"
-)]
-#[test_case(
-    [Uint128::from(0u128), Uint128::from(3564u128)],
-    [Uint128::from(3450765745u128), Uint128::from(12282531965699u128)],
-    false;
-    "Test 8: Amount of asset less than one microunit of other asset"
+// Test 8: empty pool. Should work for stable and concentrated pools, but not for xyk pools.
```

Comment:
move comment to line 277?

Reply:
why? the test_case there already has a description string. I only put these comments because the test_matrix does not allow a description string